### PR TITLE
implementing handler to deal with enconded gzip body

### PR DESCRIPTION
**Pull request by Rui-Pedro-Nunes**
This new PR is to implement a handler to deal with encoded bodies.

How it works:
  - In case of the body be encoded, it will be automatically decoded and available as usual, in the body variable.

### CODE REVIEW

File: lib/segregate.rb [hunk 1]
```diff
@@ -236,14 +251,27 @@ def parse_body line
     @stashed_body = ""
 
     if line.length >= headers['content-length'].to_i
-      @body = line
+      line = line[0..-3]
```
Comment (benSlaughter) [inline lib/segregate.rb:255]: Why do we remove the last few characters?

**Review by benSlaughter — COMMENTED** [review]
Jus one minor question, LGTM 👍

File: lib/segregate.rb (hunk 1 above)
Comment (Rui-Pedro-Nunes) [reply lib/segregate.rb:255]: I am removing the two delimiters characters from the HTTP protocol. I can add a comment to be clear.

Comment (benSlaughter) [reply lib/segregate.rb:255]: Yeah, I think that would be a good idea

**Review by benSlaughter — APPROVED** [review]
Just the one comment LGTM 👍

**Comment by Rui-Pedro-Nunes** [comment]
@benSlaughter , done. You can push the button if you are happy.

**Pull request merged**